src/main/java/com/mmm/his/cer/utility/farser/ast/node/operator/set/SetLogicOperator.java

```diff
@@ -0,0 +1,36 @@
+package com.mmm.his.cer.utility.farser.ast.node.operator.set;
+
+import com.mmm.his.cer.utility.farser.ast.node.nonterminal.SetNonTerminal;
+import com.mmm.his.cer.utility.farser.ast.node.type.LookupContext;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Base node for performing Set logic. This will ensure that the left and right nodes are processed
+ * so that data is fetched, and then perform the "math" as defined in the subclass.
+ *
+ * @author Mike Funaro
+ */
+public abstract class SetLogicOperator<C extends LookupContext<T>, T> extends
```

Comment:
Nitpick: Why "Set" but it uses "List"?

Now after looking at the other implementations, the mention of "Set" probably just means "some set of values" and does not refer to the used data type. Maybe clarify that in the javadocs at least. I was also not sure at first if it means it is an operator which "sets some data somewhere".

Reply:
Yeah this was loosely based off Set _theory_. I could change the name to reflect that? Maybe it would be more clear. `SetTheoryOperator`?

Reply:
Renamed to SetTheory where applicable.